### PR TITLE
Pair programming with Erdem, Andrew & Christy

### PR DESCRIPTION
**Pull request by christylaguardia**
#### Single-line Summary
**Today, Andrew, Erdem and Christy paired together. It took about _**

#### Reflect and summarize on your process for each `TODO` item :  
  1. First, we started on adding our connection string to server.js
  2. Next, we added the IIFI to article.js
  3. Then, we worked on Article.loadAll, Article.numWordsAll, Article.allAuthors and Article.numWordsByAuthor
  4. Then we changed the script tag in index.html
  5. Next, we added the IIFI to articleView.js
  6. We added the handelbars compile function to articleView.initAdminPage
  7. Then we changed the script tag in admin.html
  8. Then we fixed the IIFIs to include module.Article = Article and module.articleView = articleView

#### Checklist (before submitting, fill in each set of square brackets with an 'x')
- [x] We have titled the Pull Request similar to our branch name (ex: 'brian-rick'). 
- [x] This PR includes commits from both myself and my partner; e.g. We followed good pair programming practices by switching driver/navigator roles.
- [x] There is no extraneous, unrelated code included in this PR.
- [x] We have summarized our `TODO:` process above.

### CODE REVIEW

File: starter-code/public/scripts/article.js [hunk 1]
```diff
@@ -2,107 +2,137 @@
...
+    console.log('data has been loaded.');
+  };
+    // DONE: Refactor this forEach code, by using a `.map` call instead, since what we are trying to accomplish
+    // is the transformation of one colleciton into another.
+
+    /* OLD forEach():
+    rawData.forEach(function(ele) {
+    Article.all.push(new Article(ele));
+  });
+  */
```
Comment (JPGothro) [inline starter-code/public/scripts/article.js]: Good work.

File: starter-code/public/scripts/article.js [hunk 2]
```diff
@@ -2,107 +2,137 @@
...
+  };
+
+  // DONE: Chain together a `map` and a `reduce` call to get a rough count of all words in all articles.
+  Article.numWordsAll = () => {
+    return Article.all.map(function(a) {
+      return a.body.split(' ').length;
+    }).reduce(function(acc, num) {
+      return acc + num;
+    },0)
+  };
```
Comment (JPGothro) [inline starter-code/public/scripts/article.js]: Good.

File: starter-code/public/scripts/article.js [hunk 3]
```diff
@@ -2,107 +2,137 @@
...
+
+  // DONE: Chain together a `map` and a `reduce` call to produce an array of unique author names.
+  Article.allAuthors = () => {
+    return Article.all.map(function(a) {
+      return a.author;
+    }).reduce(function(uniqueAuthors, ele) {
+      if (uniqueAuthors.indexOf(ele) < 0) uniqueAuthors.push(ele);
+      return uniqueAuthors;
+    }, []);
+  };
```
Comment (JPGothro) [inline starter-code/public/scripts/article.js]: Nice.

File: starter-code/public/scripts/article.js [hunk 4]
```diff
@@ -2,107 +2,137 @@
...
+          .map(function(a) {
+            return a.body.split(' ').length;
+          })
+          // get a total count of words in all articles for that author
+          .reduce(function(acc, num) {
+            return acc + num;
+          },0)
+      }
+    })
+  };
```
Comment (JPGothro) [inline starter-code/public/scripts/article.js]: Good job.

File: starter-code/public/scripts/articleView.js [hunk 5]
```diff
@@ -1,120 +1,127 @@
 'use strict';
 
-// TODO: Wrap the entire contents of this file in an IIFE.
-// Pass in to the IIFE a module, upon which objects can be attached for later access.
+// DONE: Wrap the entire contents of this file in an IIFE.
+// Pass in to the IIFE a module, upon which objects can be attached for later access
+
+(function(module) {
 var articleView = {};
```
Comment (JPGothro) [inline starter-code/public/scripts/articleView.js]: When the IIFE was created, this line should have been indented...

File: starter-code/server.js [hunk 6]
```diff
@@ -8,10 +8,15 @@ const bodyParser = require('body-parser');
...
+// USERNAME: postgres
+// PASSWORD: miyagi
+// HOST: localhost
+// PORT: 5432
+// DATABASE: demo
+const conString = 'postgres://postgres:miyagi@localhost:5432/demo'; // DONE: Don't forget to set your own conString
 const client = new pg.Client(conString);
 client.connect();
-client.on(console.error);
+client.on('error', console.error);
```
Comment (JPGothro) [inline starter-code/server.js]: YES! An explanatory string of what's appearing in the log! GREAT!

File: starter-code/public/scripts/articleView.js [hunk 7]
```diff
@@ -1,120 +1,127 @@
...
+    // REVIEW: We use `forEach` here because we are relying on the side-effects of the callback function:
+    // appending to the DOM.
+    // The callback is not required to return anything.
+    Article.numWordsByAuthor().forEach(stat => $('.author-stats').append(template(stat)));
+    console.log('author stats template added');
+    // REVIEW: Simply write the correct values to the page:
+    $('#blog-stats .articles').text(Article.all.length);
+    $('#blog-stats .words').text(Article.numWordsAll());
+  };
+  console.log('admin page loaded');
```
Comment (JPGothro) [inline starter-code/public/scripts/articleView.js]: Shouldn't this console log be inside the function above it?

**Comment by kbrumer** [comment]
It's an IIFE, not IIFI. **I**mmediately **I**nvoked **F**unction **E**xpression. It is pronounced IF-ee.

**Pull request closed without merging**